airflow/kubernetes/istio.py

```diff
@@ -137,16 +139,19 @@ def _post_quitquitquit(self, pod, container, status_port):
         """ Send the curl to shutdown the isto-proxy container
         """
         # Use exec to curl localhost inside of the sidecar.
-        _ = stream(
-            self._client.connect_get_namespaced_pod_exec,
-            pod.metadata.name,
-            pod.metadata.namespace,
-            tty=False,
-            stderr=True,
-            stdin=False,
-            stdout=True,
-            container=container.name,
-            command=[
-                '/bin/sh',
-                '-c',
-                'curl -XPOST http://127.0.0.1:{}/quitquitquit'.format(status_port)])
+        try:
+            _ = stream(
+                self._client.connect_get_namespaced_pod_exec,
+                pod.metadata.name,
+                pod.metadata.namespace,
+                tty=False,
+                stderr=True,
+                stdin=False,
+                stdout=True,
+                container=container.name,
+                command=[
+                    '/bin/sh',
+                    '-c',
+                    'curl -XPOST http://127.0.0.1:{}/quitquitquit'.format(status_port)])
+        except ApiException:
+            self.log.warning("Istio proxy already shut down.")
```

Comment:
in your approach, you just check if it was shutdown, then don't call again, which I think could be a better approach than handling the exception like this

Reply:
Hi @sjmiller609, yeah I'm not a huge fan of creating exceptions for normal behavior.